### PR TITLE
Simple Docker CLI for Blockstack

### PR DESCRIPTION
**Pull request by jackzampolin**
```bash
$ ./bsdocker.sh 
bsdocker commands:
  start -> start the blockstack api server
  stop -> stop the blockstack api server
  setup -> run the setup for blockstack and generate a wallet
  logs -> access the logs from the blockstack api server
  build -> build the docker image for running the api server
  enter -> exec into the running docker container
```

Meant to be a wrapper around the standard docker commands that people will use. There is an inability to use the host `blockstack` cli normally because these paths get set when creating the docker installation:

```
subdomains_db = /root/.blockstack/subdomains.db
spv_path = /root/.virtualchain-spv-headers.dat
queue_path = /root/.blockstack/queues.db
```

To use a cli connected to the reciently spun up API instance use the `./bsdocker enter` command and use the `blockstack` cli in the container.

### CODE REVIEW

**Comment by kantai** [comment]
This is a pretty great CLI experience @jackzampolin 

I added 2 lines to the setup phase which make the started API endpoint bind to `0.0.0.0`, so you can now access the endpoints from the host:

```
$ curl -H 'origin: http://localhost:3000' http://localhost:6270/v1/ping
{"status": "alive", "version": "0.14.4"}
```

**Comment by jackzampolin** [comment]
Getting the hang of the docker I see @kantai! Nice init containers! I'm going to update this PR to take into account the discussion over https://github.com/blockstack/blockstack-core/pull/573 and change the dockerfile

**Comment by jackzampolin** [comment]
I've updated the docker file to use the quay.io build by default so users don't have to build locally.

**Comment by jackzampolin** [comment]
Related to: https://github.com/blockstack/blockstack-browser/pull/696

**Comment by jackzampolin** [comment]
Ok this is ready for testing! I've completed all of the TODOs from this pull: https://github.com/blockstack/blockstack-core/pull/573 and updated the readme!

File: launcher [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+push () {
+  echo "pushing build container up to quay.io..."
+  docker push $coreimage
+}
+
+commands () {
+  cat <<-EOF
+bsdocker commands:
```
Comment (larrysalibra) [inline launcher]: can we update this to say launcher instead of bsdocker?

**Comment by jackzampolin** [comment]
@larrysalibra Took care of that.

File: launcher [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+start-dev () {
+  local coretag=$gitrev
+  start $1
+}
+
+start () {
+  # Check for args first
+  if [ $# -eq 0 ]; then
+    echo "Need to input password for wallet located in the $HOME/.blockstack folder when staring api: ./launcher start mypass"
```
[the remark this answers is not in the document]
Comment (jackzampolin) [reply launcher]: Whats the typo?

[the remark this answers is not in the document]
Comment (larrysalibra) [reply launcher]: "staring api" should be "starting api"?

[the remark this answers is not in the document]
Comment (jackzampolin) [reply launcher]: 🤦‍♂️ yes. Looks like this got merged. I'll fix this in another PR.

**Pull request merged**